lib/pharos/configuration/host.rb

```diff
@@ -15,7 +15,7 @@ class Host < Pharos::Configuration::Struct
       attribute :private_address, Pharos::Types::Strict::String.optional.default(nil)
       attribute :private_interface, Pharos::Types::Strict::String.optional.default(nil)
       attribute :role, Pharos::Types::Strict::String
-      attribute :labels, Pharos::Types::Strict::Hash
+      attribute :labels, Pharos::Types::Strict::Hash.default(proc { {} })
```

Comment:
Is this the minimal fix? Other changes are just extra?

Reply:
My understanding is that setting node labels in `def labels` is the problem. The changes to `initialize` make the host set the node role labels and taints already upon initialization.